core/src/main/java/org/mule/processor/EndpointTransactionalInterceptingMessageProcessor.java

```diff
@@ -36,7 +38,7 @@ public MuleEvent process(final MuleEvent event) throws MuleException
         {
             return event;
         }
-        else
+        else if(TransactionCoordination.getInstance().getTransaction() != null)
```

Comment:
I'm not sure about this, but seems to me that if EndpointTransactionalInterceptingMessageProcessor is used, then is because there is a transaction in place. If my assumption is correct, the right change would be to avoid using this class.